src/diffusers/pipelines/stable_diffusion_3/pipeline_stable_diffusion_3.py

```diff
@@ -205,6 +205,7 @@ def _get_t5_prompt_embeds(
         self,
         prompt: Union[str, List[str]] = None,
         num_images_per_prompt: int = 1,
+        max_sequence_length: int = 256,
```

Comment:
The context training tokens size for all T5 models is 512 tokens, not 256.

Reply:
Refer to: https://github.com/huggingface/transformers/issues/5204

Reply:
yeah I know, this is still a draft. There was a discussion about if this introduced latency in the generations. It doesn't seem so but I need to test with batches yet. If it doesn't I can set it as 512, but most people won't go over 256, 512 is a lot of text.

Reply:
I have a pretty significant number of captions in my datasets longer than 256 tokens. And in comparison to the compute spent on sampling images the extra tokens are not significant.

Reply:
I did some tests and it takes longer to generate with the higher token limit, but this is only noticeable when doing batches.

prompt  (160 tokens):
> A whimsical and creative image depicting a hybrid creature that is a mix of a waffle and a hippopotamus. This imaginative creature features the distinctive, bulky body of a hippo, but with a texture and appearance resembling a golden-brown, crispy waffle. The creature might have elements like waffle squares across its skin and a syrup-like sheen. It’s set in a surreal environment that playfully combines a natural water habitat of a hippo with elements of a breakfast table setting, possibly including oversized utensils or plates in the background. The image should evoke a sense of playful absurdity and culinary fantasy.

num_images_per_prompt = 4

with 77 (like the current one):
![image](https://github.com/huggingface/diffusers/assets/5442875/1c931381-0080-4a73-8c61-4bcab3f3ac39)

with 150:
![image](https://github.com/huggingface/diffusers/assets/5442875/9bef467a-73c0-49ab-b8ba-9ee40d3a5005)

with 256:
![image](https://github.com/huggingface/diffusers/assets/5442875/3c329f6e-939f-4e09-9f25-0b3763a78045)

with 512:
![image](https://github.com/huggingface/diffusers/assets/5442875/d1c681ec-2430-468f-97a9-0a85f8313bd2)

As a reference, this text is 505 tokens:

> Imagine a whimsical and creative image that brings together the culinary delight of a breakfast waffle and the robust charm of a hippopotamus. This hybrid creature, which we’ll call the “Wafflopotamus”, is a testament to the playful absurdity of imagination. The Wafflopotamus features the distinctive, bulky body shape of a hippo, complete with its wide-open mouth and powerful jaws. However, instead of the usual grey skin, the creature’s body resembles a golden-brown, crispy waffle fresh off the griddle. The skin is textured with the familiar grid pattern of a waffle, each square perhaps filled with a glistening sheen of syrup that catches the light, adding a sense of delicious surrealism. The environment around the Wafflopotamus is as surreal as the creature itself. It combines the natural water habitat of a hippo with elements of a breakfast table setting. Picture a river of warm, melted butter in which the Wafflopotamus happily wallows, with oversized utensils or plates peeking out from the lush, pancake-like foliage in the background. Perhaps there’s even a towering pepper mill standing in for a tree, or a napkin cloth mountain range. The image is not just a visual feast but also a narrative one, telling a story of a creature that is as delightful to the taste buds as it is to the eyes. It evokes a sense of playful absurdity and culinary fantasy, inviting the viewer to step into a world where breakfast and wildlife collide in the most unexpected and amusing ways. This is the world of the Wafflopotamus, a place where the ordinary meets the extraordinary, and where imagination is the most delicious meal of all. Enjoy your visit!. As the sun rises in this fantastical world, it casts a warm, buttery glow over the scene, illuminating the Wafflopotamus in all its syrupy splendor. The creature, content in its butter river, lets out a yawn, revealing waffle teeth that are perfect squares. Nearby, a flock of silver spoon birds take flight

And this is the image it generates:

![20240613195139_569754043](https://github.com/huggingface/diffusers/assets/5442875/ca6940d4-7459-451f-80f9-c591c611aba0)

I noticed that with a big prompt it doesn't grab a lot of what is in the middle which is common in LLMs. In this prompt it doesn't generate the butter river or the elements of a breakfast table setting.

I almost never use batch generation so I don't have a strong opinion about this. @sayakpaul @yiyixuxu WDYT should be the default?

Reply:
it is a variable for pipeline `__call__` that user can adjust based on their use case; I think `256` is a pretty sensible choice:)

Reply:
also pixart sets at 300

Reply:
Setting it to 256 will affect the training example too, depending on the data the user inputs, so it needs to be an option there as well. This bit me a bunch of time times training pixart with longer prompts because all the pixart pipelines had a magic number for the token length.

Reply:
cc @sayakpaul :)

Reply:
Should be good with #8564.